src/runtime.js

```diff
@@ -67,6 +68,11 @@ export function adopt(global) {
         name: name,
         implementationClass: implementationClass
       });
+      // Resolve this extension's Service Promise.
+      getService(global, name, () => {
```

Comment:
What do you mean by "emulate"? This is so that other code can be alerted when this extension has loaded because they're waiting on the service promise to be resolved.